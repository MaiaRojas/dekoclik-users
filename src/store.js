import { createStore, applyMiddleware, compose } from 'redux';
import { reactReduxFirebase } from 'react-redux-firebase';
import { reduxFirestore } from 'redux-firestore';
import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import reducers from './reducers';
import apiMiddleware from './middleware';


firebase.initializeApp({
  apiKey: process.env.FIREBASE_API_KEY,
  authDomain: `${process.env.FIREBASE_PROJECT}.firebaseapp.com`,
  databaseURL: `https://${process.env.FIREBASE_PROJECT}.firebaseio.com`,
  projectId: process.env.FIREBASE_PROJECT,
  storageBucket: `${process.env.FIREBASE_PROJECT}.appspot.com`,
  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
});


const firestore = firebase.firestore();
firestore.settings({ timestampsInSnapshots: true });


const middlewares = [apiMiddleware];


// Add redux Firebase to compose
const createStoreWithFirebase = compose(
  reactReduxFirebase(firebase, {
    userProfile: 'users',
    useFirestoreForProfile: true,
  }),
  reduxFirestore(firebase),
)(createStore);


// if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  // const { logger } = require('redux-logger');
  // middlewares.push(logger);
// }


export default createStoreWithFirebase(reducers, applyMiddleware(...middlewares));
