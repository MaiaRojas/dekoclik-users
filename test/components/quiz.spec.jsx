import React from 'react';
import { render } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../src/store';
import Quiz from '../../src/components/quiz';


describe('<Quiz />', () => {

  it.skip('should print warning when missing part, progress, firebase, auth or match', () => {
    const spy = jest.spyOn(console, 'error').mockImplementation(jest.fn());
    expect(() => render(<Provider store={store}><Quiz /></Provider>))
      .toThrow(/startedAt/);

    expect(spy.mock.calls.length).toBe(5);
    expect(spy.mock.calls).toMatchSnapshot();

    spy.mockReset();
    spy.mockRestore();
  });

  // cuando props.progress es un objeto vacío no hay progreso
  // cuando props.progress es un array es que ya tenemos respuestas
  // cuando props.progress es un objeto con propiedad `results` es que ya se envío

  it.skip('should show notice before starting quiz', () => {});

  it.skip('should start quiz...', () => {});

  it.skip('should save progress as we go', () => {});

  it.skip('should submit quiz...', () => {});

  it.skip('should...', () => {
    const component = render((
      <Provider store={store}>
        <Quiz
          part={{
            durationString: '10min',
            questions: [
              {
                title: 'Foo',
                description: 'Blah blah blaf',
                answers: [
                  'an answer',
                  'another possible answer',
                ],
                solution: [1],
              },
            ],
          }}
          progress={{}}
          firebase={{ database: () => {} }}
          auth={{}}
          match={{
            params: {
              cohortid: 'foo',
              courseid: 'test',
              unitid: '01-bar',
              partid: '05-quiz',
            },
          }}
        />
      </Provider>
    ));
    console.log(component.html());
  });

  // it('should render to a <header> containing an <h2> with the title', () => {
  //   const component = render(<TopBar title="Blah blah blah" />);
  //   const $header = component.children().first();
  //   expect($header.get(0).tagName).toBe('header');
  //   expect($header.hasClass('mui-fixed')).toBe(true);
  //   expect(component.find('h2').text()).toBe('Blah blah blah');
  // });
  //
  // it('should have classes prop injected and title prop on instance', () => {
  //   const component = shallow(<TopBar title="foo" />);
  //   expect(component).toHaveLength(1);
  //   expect(component.props()).toMatchSnapshot()
  //   expect(component.instance().props).toEqual({ title: 'foo' });
  // });

});
