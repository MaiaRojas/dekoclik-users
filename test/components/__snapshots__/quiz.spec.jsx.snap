// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quiz /> should print warning when missing part, progress, firebase, auth or match 1`] = `
Array [
  Array [
    "Warning: Failed prop type: The prop \`part\` is marked as required in \`Quiz\`, but its value is \`undefined\`.
    in Quiz
    in WithStyles(Quiz)
    in Connect(WithStyles(Quiz))
    in Provider",
  ],
  Array [
    "Warning: Failed prop type: The prop \`progress\` is marked as required in \`Quiz\`, but its value is \`undefined\`.
    in Quiz
    in WithStyles(Quiz)
    in Connect(WithStyles(Quiz))
    in Provider",
  ],
  Array [
    "Warning: Failed prop type: The prop \`firebase\` is marked as required in \`Quiz\`, but its value is \`undefined\`.
    in Quiz
    in WithStyles(Quiz)
    in Connect(WithStyles(Quiz))
    in Provider",
  ],
  Array [
    "Warning: Failed prop type: The prop \`auth\` is marked as required in \`Quiz\`, but its value is \`undefined\`.
    in Quiz
    in WithStyles(Quiz)
    in Connect(WithStyles(Quiz))
    in Provider",
  ],
  Array [
    "Warning: Failed prop type: The prop \`match\` is marked as required in \`Quiz\`, but its value is \`undefined\`.
    in Quiz
    in WithStyles(Quiz)
    in Connect(WithStyles(Quiz))
    in Provider",
  ],
]
`;
